assets/js/components/Dashboard/Dashboard.tsx

```diff
@@ -52,23 +55,35 @@ const Dashboard: ComponentType = () => {
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
   // Fetch alerts every 4 seconds.
-  useInterval(() => {
-    fetchAlerts().then(
-      ({
-        all_alert_ids: allAPIalertIds,
-        alerts: newAlerts,
-        screens_by_alert: screensByAlertMap,
-      }) => {
-        findAndSetBannerAlert(alerts, newAlerts);
-        dispatch({
-          type: "SET_ALERTS",
-          alerts: newAlerts,
-          allAPIAlertIds: allAPIalertIds,
-          screensByAlertMap: screensByAlertMap,
+  useInterval(
+    () => {
+      fetchAlerts()
+        .then(
+          ({
+            all_alert_ids: allAPIalertIds,
+            alerts: newAlerts,
+            screens_by_alert: screensByAlertMap,
+          }) => {
+            findAndSetBannerAlert(alerts, newAlerts);
+            dispatch({
+              type: "SET_ALERTS",
+              alerts: newAlerts,
+              allAPIAlertIds: allAPIalertIds,
+              screensByAlertMap: screensByAlertMap,
+            });
+          },
+        )
+        .catch((response: Response) => {
+          if (response.status === 403) {
+            setIsAlertsIntervalRunning(false);
+            setShowModal(true);
```

Comment:
It seems we'd only show this modal if, specifically, a background alerts request fails. How is the experience if some other kind of "foreground" request fails? Should we try to integrate this into all instances where we fetch data from the server?

Reply:
Because this task was already kind of scope creep-y, I focused specifically on what I knew to be the problem we see in Sentry.

You do raise a good point though. We've had a habit of assuming a request will be successful so we unintentionally fail silently in some areas. For example, I hardcoded a 500 for the `Associate with alert` page and the only indicator that something went wrong is that the table is empty. Because this would be a bigger task, I'll get it added to the backlog for us to talk about in the future.

Reply:
Did you end up writing this task? (sorry for letting this review drop off)

Reply:
Just wrote it up [here](https://app.asana.com/0/1185117109217422/1208769995914082/f).